src/main/java/org/folio/edge/core/security/SecureStore.java

```diff
@@ -10,8 +13,27 @@ protected SecureStore(Properties properties) {
     this.properties = properties;
   }
 
+  /**
+   * Fetch a value from the store.
+   *
+   * <p>In Vert.x based code use {@code #lookup(String, String, String) to avoid blocking the event loop.
+   */
   public abstract String get(String clientId, String tenant, String username) throws NotFoundException;
 
+  public Future<String> lookup(String clientId, String tenant, String username) {
```

Comment:
I'd prefer to have an API that takes the vertx parameter as well. Perhaps both this and the other calling each other?